server/queue/queue-activemq/src/main/java/org/apache/james/queue/activemq/ActiveMQConsumerOptions.java

```diff
@@ -0,0 +1,117 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.activemq;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import org.apache.james.queue.jms.ConsumerOptions;
+
+/**
+ * The ActiveMQ specific consumer options.
+ * <p>
+ * See <a href="http://activemq.apache.org/destination-options.html">http://activemq.apache.org/destination-options.html</>
+ * for more details.
+ */
+public final class ActiveMQConsumerOptions implements ConsumerOptions {
+    private final Optional<String> dequeueParams;
```

Comment:
Sun Java guidelines (that we mostly implement with checkstyle) states that static members should come before any non-static member (true for innerclass, methods, etc)